src/hotspot/share/services/attachListener.cpp

```diff
@@ -192,7 +312,8 @@ static jint get_agent_properties(AttachOperation* op, outputStream* out) {
 // JVMTI environment that has enabled this event. However it's useful to
 // trigger the SIGBREAK handler.
 
-static jint data_dump(AttachOperation* op, outputStream* out) {
+static jint data_dump(AttachOperation* op, attachStream* out) {
+  out->set_result(JNI_OK); // allow streaming output
```

Comment:
I'm wondering, do you think it's useful to add a helper `start_streaming_if_available` (or whatever) that just calls `set_result(JNI_OK)`? That helper could have a small snippet of documentation to explain what's going on. That would probably be neater than a comment.